apps/frontend/src/app/pages/profile/profile.component.ts

```diff
@@ -96,7 +96,6 @@ export class ProfileComponent implements OnInit, OnDestroy {
               });
             }
           }
-          this.isLocal = false;
```

Comment:
`isLocal` should be getting set to `true`. The logic is a bit convoluted by the above route param checks are determining if the logged in user's page or not, and setting `isLocal` accordingly, there's just no `else` block due to early return.

Reply:
Is `isLocal` not already initialized to `true` [here](https://github.com/momentum-mod/website/blob/c7a72027ad0e8f8f23ae2d04f61263c37bb5f083/apps/frontend/src/app/pages/profile/profile.component.ts#L68C3-L68C18)?

Reply:
It's initialised as true, but Angular will keep the instance of the component alive across multiple router changes, hence we're updating this in paramMap. If a user were to navigate to another user's profile, then back to their own, isLocal would be false when it should be true

Reply:
Ahh I see, my angular component lifecycle knowledge is a bit rusty. Fixed in latest revision

Reply:
Yeah, I find the lifecycle stuff confusing sometimes, I'm not exactly sure when Angular will or won't reuse a behaviour.

But assuming that it *will* reuse a component instance, and since router events fire on first load, router events should handle every case we care about and thus more reliable than lifecycle events.